Mathlib/Analysis/Calculus/VectorField.lean

```diff
@@ -107,6 +107,62 @@ lemma lieBracket_smul_right {c : 𝕜} (hW : DifferentiableAt 𝕜 W x) :
   simp only [← differentiableWithinAt_univ, ← lieBracketWithin_univ] at hW ⊢
   exact lieBracketWithin_smul_right hW uniqueDiffWithinAt_univ
 
+/--
+Product rule for Lie Brackets: given two vector fields `V W : E → E` and a function `f : E → 𝕜`,
+we have `[V, f • W] = (df V) • W + f • [V, W]`
+-/
+lemma lieBracketWithin_fmul_right {f : E → 𝕜}
+    (hf : DifferentiableWithinAt 𝕜 f s x)
+    (hW : DifferentiableWithinAt 𝕜 W s x)
+    (hs: UniqueDiffWithinAt 𝕜 s x) :
+    lieBracketWithin 𝕜 V (fun y => f y • W y) s x =
```

Comment:
Mathlib style slightly prefers ↦ (typed with \maps or \mapsto) over => for anonymous functions.
```suggestion
    lieBracketWithin 𝕜 V (fun y ↦ f y • W y) s x =